plugin/evm/block.go

```diff
@@ -102,17 +155,84 @@ func (b *Block) syntacticVerify() error {
 
 // Verify implements the snowman.Block interface
 func (b *Block) Verify(context.Context) error {
-	return b.verify(true)
+	return b.verify(nil, true)
+}
+
+// ShouldVerifyWithContext implements the block.WithVerifyContext interface
+func (b *Block) ShouldVerifyWithContext(context.Context) (bool, error) {
```

Comment:
where do we use this?

Reply:
This is directly called by the engine to determine which Verify fn should be called